### ISSUE TITLE
Incremental Adoption

### ISSUE THREAD

**Issue by danieleades** [pre-commit/mirrors-mypy#46]
This is much a `mypy` question as a `pre-commit` question.

Mypy is very challenging to apply to large, existing codebases. The very high number of errors prevents using mypy as a quality gate on *all* code, since updating type annotations for the entire codebase in one go is simply not practical.

What i'm finding is that it's difficult to use Mypy for incremental adoption of typing in a codebase.

I can use a pre-commit hook to run Mypy on changed files with no issue, but if i'm using the [pre-commit github action](https://github.com/pre-commit/action) in CI, that's going to run on *all* files, not only on *changed* files.

An alternative might be to somehow track the total number of Mypy errors in the codebase, and require that they never *increase* as a result of changes. If there's any way to do this using pre-commit, i've not found any mention of it.

Basically i'm finding Mypy to be, in practice, a little "all or nothing".

Any ideas?

**Comment by asottile** [comment]
this isn't really the right place to ask since this repository is just a mirror

but to answer anyway:

- you can utilize pre-commit's `exclude` to opt in portions of your codebase
- you can use mypy's `ignore` to ignore errors in portions of your codebase

then you decrease the blocklists until you've covered your whole codebase

**Comment by danieleades** [comment]
> this isn't really the right place to ask since this repository is just a mirror
> 
> but to answer anyway:
> 
> * you can utilize pre-commit's `exclude` to opt in portions of your codebase
> * you can use mypy's `ignore` to ignore errors in portions of your codebase
> 
> then you decrease the blocklists until you've covered your whole codebase

thankyou anyway.

i've used the above, in addition to the `--follow-imports=skip` arg
